precompile/allowlist/role.go

```diff
@@ -77,18 +80,18 @@ func (r Role) Hash() common.Hash {
 	return common.Hash(r)
 }
 
-func (r Role) GetSetterFunctionName() string {
+func (r Role) GetSetterFunctionName() (string, error) {
 	switch r {
 	case AdminRole:
-		return "setAdmin"
+		return "setAdmin", nil
 	case ManagerRole:
-		return "setManager"
+		return "setManager", nil
 	case EnabledRole:
-		return "setEnabled"
+		return "setEnabled", nil
 	case NoRole:
-		return "setNone"
+		return "setNone", nil
 	default:
-		panic("unknown role")
+		return "", ErrInvalidRole
```

Comment:
seems this err is ignored if it is returned?

Reply:
we handle the err in case we call it with an arbitrary role. I.e calls like role.GetSetterFunctionName without a defined Role type Admin/Enabled etc.